test/integration/framework/executor/framework.go

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package executor
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"net/url"
+	"strings"
+
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	"github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/remotecommand"
+)
+
+// Executor allows for execution of commands in containers.
+type Executor interface {
+	ExecCommandInContainerWithFullOutput(ctx context.Context, namespace, podName, containerName string, cmd ...string) (stdout, stderr string, err error)
+	ExecCommandInContainer(ctx context.Context, namespace, podName, containerName string, cmd ...string) (stdout string)
+	ExecShellInContainer(ctx context.Context, namespace, podName, containerName string, cmd string) (stdout string)
+}
+
+// ExecOptions passed to ExecWithOptions
+type ExecOptions struct {
```

Comment:
The `ExecOptions` really should only be options and not contain things that are essential to the functioning like the `Client`. Also, as @ialidzhikov said, this is only used internally, so try to trim it down and move as many things to the `executor` as possible.